pysilcam/silcamgui/interactive_summary.py

```diff
@@ -461,76 +500,88 @@ def toggle_plot(self):
 
     def update_plot(self, save=False):
         '''update the plots and save to excel is save=True'''
+
+        vd_total = xr.DataArray(data=self.vd_total, dims=['time', 'particle_size'], coords=[self.u, self.dias])
+        vd_gas = xr.DataArray(data=self.vd_gas, dims=['time', 'particle_size'], coords=[self.u, self.dias])
+        vd_oil = xr.DataArray(data=self.vd_oil, dims=['time', 'particle_size'], coords=[self.u, self.dias])
+
+        ds = xr.Dataset()
+        ds['vd_total'] = vd_total
+        ds['vd_gas'] = vd_gas
+        ds['vd_oil'] = vd_oil
+
         start_time = self.mid_time - self.av_window / 2
         end_time = self.mid_time + self.av_window / 2
-        u = pd.to_datetime(self.u)
-        timeind = np.argwhere((u >= start_time) & (u < end_time))
-
-        psd_nims = len(timeind)
+        
+        ds_slice = ds.sel(time=slice(start_time, end_time))
+        psd_nims = ds_slice.time.size
         if psd_nims < 1:
-            plt.sca(self.axispsd)
-            plt.cla()
+            psd = ds_slice.mean(dim='time')
+            #plt.sca(self.axispsd)
+            self.axispsd.clear()
 
-            plt.sca(self.axistext)
+            #plt.sca(self.axistext)
 
             string = ''
             string += '\n Num images: {:0.0f}'.format(psd_nims)
-            string += '\n Start: ' + str(start_time.tz_convert(None))
-            string += '\n End: ' + str(end_time.tz_convert(None))
+            string += '\n Start: ' + str(start_time)
+            string += '\n End: ' + str(end_time)
             string += '\n Window [sec.]: {:0.3f}'.format((end_time - start_time).total_seconds())
 
-            plt.title(string, verticalalignment='top', horizontalalignment='right', loc='right')
+            self.axistext.clear()
+            self.axistext.text(1, 1, string, va='top', ha='right', transform=self.axistext.transAxes)
+            self.axistext.axis('off')
 
-            plt.sca(self.axisconstant)
+            #plt.sca(self.axisconstant)
```

Comment:
E265: block comment should start with '# '  <!-- Automatically posted by Lintly -->